catalyst/rl/offpolicy/discrete_sampler.py

```diff
@@ -0,0 +1,256 @@
+import os
+import time
+import copy
+import random
+import numpy as np
+import torch
+from datetime import datetime
+from tensorboardX import SummaryWriter
+
+from catalyst.utils.misc import set_global_seeds
+from catalyst.dl.utils import UtilsFactory
+from catalyst.utils.serialization import serialize, deserialize
+from catalyst.rl.random_process import RandomProcess
+from catalyst.rl.offpolicy.utils import SamplerBuffer
+
+# speed up optimization
+os.environ["OMP_NUM_THREADS"] = "1"
+torch.set_num_threads(1)
+SEED_RANGE = 2**32 - 2
+
+
+def get_critic_weights(critic, exclude_norm=False):
+    state_dict = critic.state_dict()
+    if exclude_norm:
+        state_dict = {
+            key: value
+            for key, value in state_dict.items()
+            if all(x not in key for x in ["norm", "lstm"])
+        }
+    state_dict = {key: value.clone() for key, value in state_dict.items()}
+    return state_dict
+
+
+def set_critic_weights(critic, weights, strict=True):
+    critic.load_state_dict(weights, strict=strict)
+
+
+class Sampler:
```

Comment:
to reduce copy-paste I suggest you to specify noise, action and buffer creation functions during the init based on env.type (`'discrete' or 'continuous'`), just like the trainer losses done.

Reply:
Agree